### PR TITLE
refactor LogCameraInfo

### PR DESCRIPTION
**Pull request by deanlee**
refactor `LogCameraInfo` to simplify the initialization of encoders according to the device type

### CODE REVIEW

File: selfdrive/loggerd/loggerd.cc [hunk 1]
```diff
@@ -44,13 +45,11 @@ namespace {
 constexpr int MAIN_FPS = 20;
 
 #ifndef QCOM2
-constexpr int MAIN_BITRATE = 5000000;
-constexpr int MAX_CAM_IDX = LOG_CAMERA_ID_DCAMERA;
-constexpr int DCAM_BITRATE = 2500000;
+#define MAIN_BITRATE 5000000
```
Comment (adeebshihadeh) [inline selfdrive/loggerd/loggerd.cc:48]: why change these to defines?

File: selfdrive/loggerd/loggerd.cc [hunk 2]
```diff
@@ -70,7 +69,8 @@ LogCameraInfo cameras_logged[LOG_CAMERA_ID_MAX] = {
     .downscale = false,
     .has_qcamera = true
   },
-  [LOG_CAMERA_ID_DCAMERA] = {
+#if defined(QCOM) || defined(QCOM2)
```
Comment (adeebshihadeh) [inline selfdrive/loggerd/loggerd.cc:72]: we should remove these ifdefs, loggerd shouldn't really care about this stuff. should be handled automatically based on what kind of frame it gets

File: selfdrive/loggerd/loggerd.cc (hunk 1 above)
Comment (deanlee) [reply selfdrive/loggerd/loggerd.cc:48]: avoid unused variables error

File: selfdrive/loggerd/loggerd.cc (hunk 2 above)
Comment (deanlee) [reply selfdrive/loggerd/loggerd.cc:72]: if so, The code will be more complicated than it is now.

**Pull request closed without merging**